airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -73,191 +73,129 @@
-        connectionUpdaterInput.setJobId(null);
-        connectionUpdaterInput.setAttemptNumber(1);
-        connectionUpdaterInput.setFromFailure(false);
-        connectionUpdaterInput.setAttemptId(null);
-      } else {
-        workflowState.setContinueAsReset(false);
+        reportCancelledAndContinueWith(true, connectionUpdaterInput);
       }
 
-      if (workflowState.isUpdated() && !workflowState.isRunning()) {
-        log.error("A connection configuration has changed for the connection {} when no sync was running. The job will be restarted",
-            connectionUpdaterInput.getConnectionId());
-      } else if (workflowState.isDeleted()) {
-        // Stop the runs
-        final ConnectionDeletionInput connectionDeletionInput = new ConnectionDeletionInput(connectionUpdaterInput.getConnectionId());
-        runMandatoryActivity((input) -> connectionDeletionActivity.deleteConnection(input), connectionDeletionInput);
-        return;
-      } else if (workflowState.isCancelled() || workflowState.isCancelledForReset()) {
-        runMandatoryActivity((input) -> jobCreationAndStatusUpdateActivity.jobCancelled(input),
-            new JobCancelledInput(
-                maybeJobId.get(),
-                maybeAttemptId.get(),
-                FailureHelper.failureSummaryForCancellation(maybeJobId.get(), maybeAttemptId.get(), failures, partialSuccess)));
-        resetNewConnectionInput(connectionUpdaterInput);
-      } else if (workflowState.isFailed()) {
-        reportFailure(connectionUpdaterInput);
-      } else {
-        // report success
-        reportSuccess(connectionUpdaterInput);
+      if (workflowState.isCancelled()) {
+        reportCancelledAndContinueWith(false, connectionUpdaterInput);
       }
-      continueAsNew(connectionUpdaterInput);
+
     } catch (final Exception e) {
       log.error("The connection update workflow has failed, will create a new attempt.", e);
-
-      reportFailure(connectionUpdaterInput);
-      continueAsNew(connectionUpdaterInput);
+      reportFailure(connectionUpdaterInput, null);
+      prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
     }
   }
 
-  private void reportSuccess(final ConnectionUpdaterInput connectionUpdaterInput) {
+  private CancellationScope generateSyncWorkflowRunnable(ConnectionUpdaterInput connectionUpdaterInput) {
+    return Workflow.newCancellationScope(() -> {
+      connectionId = connectionUpdaterInput.getConnectionId();
+
+      if (connectionUpdaterInput.getWorkflowState() != null) {
+        workflowState = connectionUpdaterInput.getWorkflowState();
+      }
+
+      workflowState.setResetConnection(connectionUpdaterInput.isResetConnection());
+
+      Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
+
+      Workflow.await(timeToWait,
+          () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+
+      val needToProcessSync = !workflowState.isUpdated() && !workflowState.isDeleted();
```

Comment:
is it possible to invert this? it would reduce the number of right brackets and make it easier to read.

```
if (!needToProcessSync) {
  return; // or continueAsNew()
}

// rest of code that is currently in needToProcessSync
```

Reply:
Done, I also delete the need to process val